From: Simon McVittie <smcv@debian.org>
Date: Wed, 17 Dec 2025 14:30:24 +0000
Subject: Tell "as-installed" tests where they can find triggers

Since commit d10e1148 "Add initial support for preinstalling flatpaks",
the test suite sets FLATPAK_DATA_DIR to a temporary directory, both
while running uninstalled and as-installed.

While running uninstalled we already set FLATPAK_TRIGGERSDIR to the
trigger scripts in the source tree, but when running "as-installed",
we need to run the triggers that the OS installs as part of the flatpak
package (or equivalent).

Not having this caused autopkgtests (automated as-installed tests) in
Debian to regress with 1.17.x.

Fixes: d10e1148 "Add initial support for preinstalling flatpaks"
Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://github.com/flatpak/flatpak/pull/6444
Applied-upstream: 1.17.3, commit:c324778b0720cb2d01c217fdc70358e232c38b4d
---
 tests/installed-tests.sh.in | 3 +++
 tests/meson.build           | 1 +
 2 files changed, 4 insertions(+)

diff --git a/tests/installed-tests.sh.in b/tests/installed-tests.sh.in
index e0978ec..765995e 100644
--- a/tests/installed-tests.sh.in
+++ b/tests/installed-tests.sh.in
@@ -1,4 +1,7 @@
 # Copyright 2024 Collabora Ltd.
 # SPDX-License-Identifier: LGPL-2.1-or-later
 
+FLATPAK_TRIGGERSDIR='@FLATPAK_TRIGGERSDIR@'
 FUSERMOUNT='@FUSERMOUNT@'
+
+export FLATPAK_TRIGGERSDIR
diff --git a/tests/meson.build b/tests/meson.build
index 852ecea..c7018b8 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -46,6 +46,7 @@ if get_option('installed_tests')
     output : 'installed-tests.sh',
     configuration : {
       'FUSERMOUNT' : fusermount,
+      'FLATPAK_TRIGGERSDIR' : get_option('prefix') / get_option('datadir') / 'flatpak' / 'triggers',
     },
     install_dir : installed_testdir,
   )
